arangod/Aql/QueryCursor.h

```diff
@@ -96,6 +96,9 @@ class QueryStreamCursor final : public arangodb::Cursor {
 
   size_t count() const override final { return 0; }
 
+  // TODO add this to Cursor and make it virtual / override final.
+  ExecutionState prepareDump(std::function<void()> const& continueHandler);
```

Comment:
Remove continueHandler here. It needs to be bound to the query before (and in fact it is)